### ISSUE TITLE
[BUG] <Blobs in G3 arcs>

### ISSUE THREAD

**Issue by MichaelJLew** [prusa3d/Prusa-Firmware-Buddy#3420]
**Printer type** - [MK4]

**Printer firmware version** - [5.1.0 alpha 2]

**Original or Custom firmware** - [Original]

**Optional upgrades** - [none]

**USB drive or USB/Octoprint**
  USB

**Describe the bug**
 G3 arcs are interrupted by small blobs that ruin the smooth arc, as shown in the photographs.

I cannot really be sure, but I suspect that the blobs might be caused by minor pauses in the movement as the arcs are being printed. 

The problem is not due to the recent fix of issue #3179 because it persisted unchanged in a modified file where there were no Z parameters in the G3 arc commands. The result with that version of the object is shown in the photograph.

**How to reproduce**
  The attached gcode file shows the issue, and is the code used to make the print in the photographs.

**Expected behavior**
  A G3 arc should be smooth and uninterrupted. The example file prints with correctly smooth curves on my RepRap firmware Duet board Voron, as shown in the photograph, and it should do the same on my MK4.

**G-code**
  Attached as a zip.

**Video**
  Please attach a video. It usually helps to solve the problem.
It's a pair of annotated photographs, not a video.
![G3 blobs pictures 001](https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/141463653/2167a64d-9d05-487c-b697-caa138400fbc)
[M10 body spirals exp h.gcode.zip](https://github.com/prusa3d/Prusa-Firmware-Buddy/files/13199715/M10.body.spirals.exp.h.gcode.zip)

**Comment by MichaelJLew** [comment]
I will add that the problem is not related to the filament, as I have printed the object with three different filaments and the blobs are consistent.

**Comment by MichaelJLew** [comment]
I have now confirmed that this issue is not restricted to G3 arcs but can be present with arcs consisting of lots of G1 lines. My explorations make it likely that the blobs happen when there are too many short line segments in the arcs. For objects sliced in PrusaSlicer that happens consistently with circular shapes with a radius of about 2mm printed in vase mode, but it occurs occasionally with non-vase mode and non-circular shapes as well. I will be putting in another bug report on this (sorry!) because this one is now wrongly titled and appears to have had no attention at all.

Printing gcode that gives these blobs on the Mk4 on a Voron using RepRap firmware gives no blobs and so the problem lies in the Prusa firmware. It is provoked by PrusaSlicer in some situations, particularly when slicing in vase mode, but the firmware should be able to deal with the number of line segments involved.

<img width="906" alt="image" src="https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/141463653/1a731e26-26d8-4c57-b2cd-f5f27cdee1e2">

Here is OpenSCAD code to make this model:
scale([1,1,1]){
    for(i=[0:45:360]){
        rotate([0,0,i])translate([5,0,0])cylinder(d=4,h=20);
    }
    cylinder(d=9.7,h=20);
}

I included the scale function in that OpenSCAD code because the blobs can be made to appear or not depending on the scale. For example, with scale(2,2,1]) there are no blobs.
